publisher/product/vocabulary/build.sh

```diff
@@ -47,56 +47,72 @@ function vocabularyGetOntologies() {
+  logItem "Merging all dev" "$(logFileName ${tag_root}/dev.owl)"
   robot merge --input "${source_family_root}/${DEV_SPEC}" --output ${TMPDIR}/dev.owl
 
   #
   # Get ontologies for Prod
   #
-  log "Merging all prod ontologies into one RDF file: : $(logFileName ${tag_root}/prod.owl)"
+  logItem "Merging all prod" "$(logFileName ${tag_root}/prod.owl)"
   robot merge --input "${source_family_root}/${PROD_SPEC}" --output ${TMPDIR}/prod.owl
   
   return 0
 }
 
+function replace_variables() {
+  sed	-e "s&ONTPUB_FAMILY&${ONTPUB_FAMILY}&g" \
+	-e "s&PRODUCT_ROOT_URL&${product_root_url}&g" \
+	-e "s&HYGIENE_TEST_PARAMETER_VALUE&${HYGIENE_TEST_PARAMETER_VALUE}&g" \
+	-e "s&ONTPUB_NAME&${ONTPUB_FAMILY^^}&g"
+}
+
 function vocabularyCreateFromOntologies() {
 
+  require ONTPUB_FAMILY || return $?
+  require product_root_url || return $?
+  require HYGIENE_TEST_PARAMETER_VALUE || return ?
   require vocabulary_script_dir || return $?
+  require vocabulary_product_tag_root || return $?
   
   logStep "vocabularyCreateFromOntologies"
-  log "Creating vocabularies form ontologies"
-  
+  log "Creating vocabularies from ontologies"
 
-  log "Creating Dev vocabulary"
-  ${JENA_ARQ} --data ${TMPDIR}/dev.owl --query ${SCRIPT_DIR}/product/vocabulary/classes.sparql --results=TTL > ${TMPDIR}/classes.ttl
-  ${JENA_ARQ} --data ${TMPDIR}/dev.owl --query ${SCRIPT_DIR}/product/vocabulary/subclasses.sparql --results=TTL > ${TMPDIR}/subclasses.ttl
-  ${JENA_ARQ} --data ${TMPDIR}/dev.owl --query ${SCRIPT_DIR}/product/vocabulary/properties.sparql --results=TTL > ${TMPDIR}/properties.ttl
-  ${JENA_ARQ} --data ${TMPDIR}/dev.owl --query ${SCRIPT_DIR}/product/vocabulary/subproperties.sparql --results=TTL > ${TMPDIR}/subproperties.ttl
-  
+  logItem "Creating scaffolding file" "$(logFileName ${tag_root}/scaffolding.ttl)"
+  rm -f "${TMPDIR}/scaffolding.ttl" ; replace_variables < "${SCRIPT_DIR}/product/vocabulary/scaffolding.ttl" > "${TMPDIR}/scaffolding.ttl"
 
+  log "Creating Dev vocabulary"
+  export data_ttl=""
+  for SPARQL in ${SCRIPT_DIR}/product/vocabulary/*.sparql ; do
+   if [ -r "${SPARQL}" ] ; then
+    ttl="${TMPDIR}/$(basename "${SPARQL}" | sed 's/.sparql/.ttl/g')" ; rm -f "${ttl}"
+    logItem "$(logFileName $(basename ${SPARQL}))" "$(logFileName ${tag_root}/$(basename ${ttl}))"
+    ${JENA_ARQ} --data ${TMPDIR}/dev.owl --query <(replace_variables < "${SPARQL}") --results=TTL > "${ttl}" && export data_ttl="${data_ttl} --data ${ttl}"
+   fi
+  done
+
+  logItem "Creating dev file" "$(logFileName ${vocabulary_product_tag_root}/${ONTPUB_FAMILY}-vD.ttl)"
   ${JENA_ARQ} \
-    --data ${SCRIPT_DIR}/product/vocabulary/scaffolding.ttl \
-    --data ${TMPDIR}/classes.ttl \
-    --data ${TMPDIR}/subclasses.ttl \
-    --data ${TMPDIR}/properties.ttl \
-    --data ${TMPDIR}/subproperties.ttl \
-    --query=/publisher/lib/echo.sparql \
+    --data "${TMPDIR}/scaffolding.ttl" \
+    ${data_ttl} \
+    --query=<(echo 'CONSTRUCT {?s ?p ?o} WHERE {?s ?p ?o}') \
```

Comment:
I do not see the point of not using echo.sparql file - it makes the code less readable. But if you want to replace reference to a file with a reference to a text, then this should be done consistently, at least across this file. So you need to do a similar replacement for https://github.com/edmcouncil/ontology-publisher/blob/7be84ac966cb422c835efe07082c22464e7639c5/publisher/product/vocabulary/build.sh#L132 and the subsequent lines.
The echo.sparql file may be removed.

Reply:
The [echo.sparql](https://github.com/edmcouncil/ontology-publisher/blob/develop/publisher/lib/echo.sparql) file is still used in the following places:

https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build.sh#L707
https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build.sh#L720
https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build.sh#L723
https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build.sh#L724
https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build-about.sh#L42
https://github.com/edmcouncil/ontology-publisher/blob/a2af17fa3bea84259325bf4f2484703892303c34/publisher/product/ontology/build-about.sh#L80